src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/components/LoseProgressModal/index.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import ConfirmDialog from "src/components/ConfirmDialog";
+
+interface Props {
+  isModalOpen: boolean;
+  setIsModalOpen(x: boolean): void;
+  onClose(): void;
+  clearState(): void;
+}
+
+export const LoseProgressModal = ({
+  isModalOpen,
+  setIsModalOpen,
+  onClose,
+  clearState,
+}: Props): JSX.Element => {
+  const title = "Leave sample editing?";
+  const message =
+    "If you leave, your current edits will be canceled and your work will not be saved.";
+
+  const handleCloseAfterConfirmation = function () {
+    // user confirms that they would like to leave edit and lose progress
+    clearState();
+    onClose();
+  };
+
+  const handleLoseProgressModalClose = function () {
+    // User wants to return to edit and not abandon changes
+    setIsModalOpen(false);
```

Comment:
this could've been part of the onClose function, instead of adding an extra prop here.